### PR TITLE
Telegram smart event notifications

### PR DESCRIPTION
**Pull request by mjdyson**
Added functionality to send notifications via Telegram. The base_url must be provided.

### CODE REVIEW

**Comment by mjdyson** [comment]
Hoping it's useful to share this small additional feature. 
The data for the notification was trickier than I expected. The Telegram service wouldnt work for me using homeassistant.local:8123, so I've had to specify the IP address directly. I'm not sure if that's a local network issue, but would be good to have other people's views on it.

File: blueprints/automation/unifiprotect/push_notification_smart_event.yaml [hunk 1]
```diff
@@ -315,6 +333,16 @@ action:
...
+  - choose:
+      - conditions: "{{ input_send_telegram }}"
+        sequence:
+          - service: notify.{{ input_notify_telegram }}
+            data:
+              title: "{{ notification_title }}"
+              message: "{{ notification_message }}"
+              data:
+                photo:
+                  url: "{{ base_url }}{{ notification_image }}"
```
Comment (AngellusMortis) [inline blueprints/automation/unifiprotect/push_notification_smart_event.yaml:345]: `notification_image` is already `input_base_image_url` + `state_attr(camera_entity_id, 'entity_picture')`

Using base URL is redundant here. Also, `base_url` is your "internal" HA URL, `base_image_url` is your "external" HA URL.

File: blueprints/automation/unifiprotect/push_notification_smart_event.yaml [hunk 2]
```diff
@@ -108,6 +109,18 @@ blueprint:
...
         text:
+    send_telegram:
+      name: (Optional) Telegram Notification
+      description: "Send a notification via Telegram"
+      default: false
+      selector:
+        boolean:
+    notify_telegram:
+      name: (Optional) Notification Target (Telegram)
+      description: "The notification target for Telegram notifications. Should be name of the Telegram bot you have configured. https://www.home-assistant.io/integrations/telegram/"
```
Comment (AngellusMortis) [inline blueprints/automation/unifiprotect/push_notification_smart_event.yaml:120]:
```suggestion
      description: >
        The notification target for Telegram notifications. Should be name of the Telegram bot you have configured.
        https://www.home-assistant.io/integrations/telegram/
```

File: blueprints/automation/unifiprotect/push_notification_smart_event.yaml [hunk 3]
```diff
@@ -16,6 +16,7 @@ blueprint:
 
       - Precense filter for only sending notifications for when you are not home
       - [HTML5 Push Notification Target][1] and/or [Mobile App Notification Target][2] as well as toggles for both
+      - Telegram Notification option, including toggle
```
Comment (AngellusMortis) [inline blueprints/automation/unifiprotect/push_notification_smart_event.yaml:19]: The description is markdown whereas the help text options below are _not_. So, you should add a link to the Telegram documentation here, similar to all of the other ones.

**Comment by AngellusMortis** [comment]
> The Telegram service wouldn't work for me using homeassistant.local:8123, so I've had to specify the IP address directly. I'm not sure if that's a local network issue, but would be good to have other people's views on it.

Yes, that is likely a network issue. I do not know like anything about Telegram, but based on how metadata scrapers work for other platforms like Twitter, Facebook, Discord, etc. Your URLs _must_ be publicly accessible. Similar to the same issue with Push Notifications. That is explicitly what the `base_image_url` option is for in the template.

File: blueprints/automation/unifiprotect/push_notification_smart_event.yaml (hunk 1 above)
Comment (mjdyson) [reply blueprints/automation/unifiprotect/push_notification_smart_event.yaml:345]: Thanks for the feedback!
Good call on this one. I'll remove base_url where it's used. 
My HA setup does not have an external IP (hence using Telegram to punch out), however, it looks like a local URL will still work with input_base_image_url.
Also for Telegram to work it's not optional.

**Comment by AngellusMortis** [comment]
It looks like you can add buttons to the Telegram message for callbacks as well. Similar to the actionable notifications for push.

https://www.home-assistant.io/integrations/telegram_bot#event-triggering
https://www.home-assistant.io/integrations/telegram/#inline_keyboard

May be something you want to add as well. Also, likely want to add this to the other 2 blueprints as well.

**Comment by mjdyson** [comment]
I will progress other files at some point today. Are the recent edits OK? My first code-based PR, apologies in advance for n00b errors.

**Comment by AngellusMortis** [comment]
Yeah, ever thing looks good now. I will have to figure out Telegram to set it up and use it. Never really had the need for it.

**Comment by AngellusMortis** [comment]
I have a number of changes/tweaks to the template. Do you mind if I close this PR and make my own? Then I can have you test it one I am done with it.

**Comment by mjdyson** [comment]
Not at all. I'll resync when you're done.

**Comment by AngellusMortis** [comment]
Closing in favor of https://github.com/briis/unifiprotect/pull/415

**Pull request closed without merging**